src/octree/mod.rs

```diff
@@ -147,41 +124,20 @@ pub struct NodeData {
     pub color: Vec<u8>,
 }
 
-fn bounding_box_to_aabb(bounding_box: &proto::AxisAlignedCuboid) -> Aabb3<f64> {
-    let min = bounding_box.min.clone().unwrap_or_else(|| {
-        let deprecated_min = bounding_box.deprecated_min.clone().unwrap(); // Version 9
-        let mut v = proto::Vector3d::new();
-        v.set_x(f64::from(deprecated_min.x));
-        v.set_y(f64::from(deprecated_min.y));
-        v.set_z(f64::from(deprecated_min.z));
-        v
-    });
-    let max = bounding_box.max.clone().unwrap_or_else(|| {
-        let deprecated_max = bounding_box.deprecated_max.clone().unwrap(); // Version 9
-        let mut v = proto::Vector3d::new();
-        v.set_x(f64::from(deprecated_max.x));
-        v.set_y(f64::from(deprecated_max.y));
-        v.set_z(f64::from(deprecated_max.z));
-        v
-    });
-    Aabb3::new(
-        Point3::new(min.x, min.y, min.z),
-        Point3::new(max.x, max.y, max.z),
-    )
-}
-
 impl Octree {
     // TODO(sirver): This creates an object that is only partially usable.
     pub fn from_data_provider(data_provider: Box<dyn DataProvider>) -> Result<Self> {
         let meta_proto = data_provider.meta_proto()?;
+        if meta_proto.version < CURRENT_VERSION {
+            println!(
+                "Data is an older octree version: {}, current would be {}. \
```

Comment:
maybe there should be point cloud instead of octree, because from version 12 also a s2 point cloud are possible